materials/internals/definition.yaml

```diff
@@ -0,0 +1,350 @@
+      color-red-95: 'hsl(339.1304347826087, 100%, 95%)'
+      color-purple: '#b866ff'
+      color-black: '#1a1a1a'
+      color-white: '#fff'
+  borderRadius:
+    label: Border Radius
+    prefix: border-radius
+    choices:
+      border-radius-1: 1px
+      border-radius-2: 2px
+      border-radius-4: 4px
+      border-radius-6: 6px
+      border-radius-20: 20px
+  shadows:
+    label: Shadows
+    prefix: shadow
+    choices:
+      shadow-1: '0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24)'
+      shadow-2: '0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23)'
+      shadow-3: '0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23)'
+      shadow-4: '0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22)'
+      shadow-5: '0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.22)'
+      shadow-6: '0 -1px 2px 0 rgba(0, 0, 0, 0.2)'
+      shadow-7: '0 -1px 1.5px 0 rgba(0, 0, 0, 0.12), 0 1px 1px 0 rgba(0, 0, 0, 0.24)'
+      shadow-8: '0 1px 9.5px 0 rgba(0, 0, 0, 0.12), 0 2px 4px 0 rgba(0, 0, 0, 0.24)'
+      shadow-9: 'inset 0 -1px 3px 0 rgba(0, 0, 0, 0.1), inset 0 1px 1.5px 0 rgba(0, 0, 0, 0.2), inset 0 1px 1.5px 0 rgba(0, 0, 0, 0.25)'
+      shadow-10: '0 0 0 1px rgba(224, 230, 237, 0.5)'
+      shadow-11: '0 1px 0.5px rgba(0, 0, 0, 0.24), 0 -1px 0.75px rgba(0, 0, 0, 0.12)'
+      shadow-12: '0 0 1px rgba(0, 0, 0, 0.25)'
+      shadow-13: '0 0 1px rgba(0, 0, 0, 0.25)'
+      shadow-14: '0 0 0.5px rgba(0, 0, 0, 0.1)'
+      shadow-15: '0 2px 2px rgba(0, 0, 0, 0.24), 0 1px 4.75px rgba(0, 0, 0, 0.12)'
+  constraints:
+    label: Constraints
+    prefix: constraint
+    choices:
+      constraint-scale: 100%
+      constraint-xs: 50px
+      constraint-s: 132px
+      constraint-m: 355px
+      constraint-l: 400px
+      constraint-xl: 768px
+  spacings:
+    label: Spacings
+    prefix: spacing
+    choices:
+      spacing-4: 4px
+      spacing-8: 8px
+      spacing-16: 16px
+      spacing-24: 24px
+      spacing-32: 32px
+  transitions:
+    label: Transitions
+    prefix: transition
+    choices:
+      transition-linear-80ms: '80ms linear'
+      transition-easeinout-150ms: '150ms ease-in-out'
+      transition-standard: 'all 0.2s ease'
+  breakpoints:
+    label: Breakpoints
+    prefix: break-point
+    choices: 
+      break-point-mobile: 768px
+      break-point-desktop: 1024px
+      break-point-biggerdesktop: 1280px
+      break-point-giantdesktop: 1680px
+      break-point-jumbodesktop: 1920px
+
+
+states:
+  hovered:
+    description: 'When the mouse is over the element'
+  # not sure we should actually have this one
+  pristine:
+    description: 'The element was not interacted with yet'
+  disabled:
+    description: 'When the element can not be interacted with'
+  readonly:
+    description: 'When the element can not be modified'
+  warning:
+    description: 'When the element has slightly wrong input'
+  error:
+    description: 'When the element has invalid input'
```

Comment:
We identified the following states:

| State       | Maps to CSS pseudo class | Description                                                                                           |
| ----------- | ------------------------ | ----------------------------------------------------------------------------------------------------- |
| `visited`   | `:visited`               |  The user has visited this link before                                                                |
| `hovered`   | `:hover`                 |  The pointer is over this element                                                                     |
| `active`    | `:active`                |  The element is currently being activated by the user                                                 |
| `focused`   | `:focus`                 | The element is currently selected to receive input                                                    |
| `disabled`  | `:disabled`              | The element can not be interacted with                                                                |
| `read-only` | `:read-only`             | The element can not be modified                                                                       |
| `checked`   | `:checked`               | The element is checked by the user (e.g. for checkboxes).                                             |
| `warning`   | -                        | The element contains a warning. Warnings do not block the further flow but should be resolved.        |
| `error`     | -                        | The element contains an error. Errors block the further flow and must be resolved.                    |
| `info`      | -                        | The element contains a hint. Hints do not block the further flow and need do not need to be resolved. |